tests/imu/imu_accelerometer_tests.cpp

```diff
@@ -0,0 +1,309 @@
+#include <gmock/gmock.h>
+#include "accelerometer.hpp"
+#include "gtest/gtest.h"
+#include "mock_i2c.hpp"
+
+using ::testing::_;
+using ::testing::NiceMock;
+using ::testing::Return;
+
+namespace {
+
+class AccelerometerTests : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    ON_CALL(*i2c_handler_, Write(_, _, _))
+        .WillByDefault(Return(types::DriverStatus::OK));
+
+    ON_CALL(*i2c_handler_, ReadContentFromRegister(_, _, _, _))
+        .WillByDefault(Return(answer_to_who_am_i_MPU9255));
+    ON_CALL(*i2c_handler_, ReadContentFromRegister(_, imu::WHO_AM_I_MPU9255_REGISTER, _, _))
+        .WillByDefault(Return(answer_to_who_am_i_MPU9255));
+    ON_CALL(*i2c_handler_, ReadContentFromRegister(_, imu::WHO_AM_I_AK8963_REGISTER, _, _))
+        .WillByDefault(Return(answer_to_who_am_i_AK8963));
+    ON_CALL(*i2c_handler_, ReadContentFromRegister(_, imu::ACCEL_CONFIG, _, _))
+        .WillByDefault(Return(answer_to_accel_config));
+    ON_CALL(*i2c_handler_, ReadContentFromRegister(_, imu::ACCEL_MEASUREMENT_DATA, _, _))
+        .WillByDefault(Return(answer_to_update));
+  }
+
+  virtual void ConfigureUnitUnderTest() {
+    unit_under_test_ = std::make_unique<imu::Accelerometer>(i2c_handler_);
+    unit_under_test_->SetSensitivity(types::ImuSensitivity::FINEST);
+  }
+
+  uint8_t i2c_address_ = imu::MPU9255_ADDRESS;
+  std::shared_ptr<i2c::MockI2C> i2c_handler_ = std::make_shared<NiceMock<i2c::MockI2C>>();
+  std::unique_ptr<imu::Accelerometer> unit_under_test_;
+
+  std::pair<types::DriverStatus, std::vector<std::uint8_t>> answer_to_who_am_i_MPU9255{
+      types::DriverStatus::OK, {imu::WHO_AM_I_MPU9255_VALUE}};
+  std::pair<types::DriverStatus, std::vector<std::uint8_t>> answer_to_who_am_i_AK8963{
+      types::DriverStatus::OK, {imu::WHO_AM_I_AK8963_VALUE}};
+  std::pair<types::DriverStatus, std::vector<std::uint8_t>> answer_to_accel_config{
+      types::DriverStatus::OK, {0b11111111}};
+  std::pair<types::DriverStatus, std::vector<std::uint8_t>> answer_to_update{
+      types::DriverStatus::OK, {0x7F, 0xFF, 0x00, 0x00, 0x80, 0x00}};
+  std::pair<types::DriverStatus, std::vector<std::uint8_t>> answer_read_mismatch{
+      types::DriverStatus::OK, {0, 15}};
+  std::pair<types::DriverStatus, std::vector<std::uint8_t>> answer_invalid{
+      types::DriverStatus::HAL_ERROR, {}};
+};
+
+TEST_F(AccelerometerTests, Init_OK) {
+  ConfigureUnitUnderTest();
+
+  auto init_return = unit_under_test_->Init(i2c_address_);
+
+  EXPECT_EQ(init_return, types::DriverStatus::OK);
```

Comment:
Why EXPECT_EQ and not ASSERT_EQ. I believe there is a gtest flag that can ignore expected results

Reply:
This is the only thing I found in your tests

Reply:
Stack Overflow says for this: 
"Use ASSERT when the condition must hold - if it doesn't the test stops right there. Use this when the remainder of the test doesn't have semantic meaning without this condition holding.

Use EXPECT when the condition should hold, but in cases where it doesn't we can still get value out of continuing the test. (The test will still ultimately fail at the end, though.)

The rule of thumb is: use EXPECT by default, unless you require something to hold for the remainder of the tests, in which case you should use ASSERT for that particular condition."

Reply:
I think in most cases "EXPECT" is good enough. A failed is a failed is a failed.